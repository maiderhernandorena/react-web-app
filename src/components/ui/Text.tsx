import React from 'react';

import Link from './Link'

type TextType = 'title' | 'subtitle' | 'link' |'p'

type TextProps = {
  message: string;
  link?: string;
  type?: TextType;
}

const Text: React.FC<TextProps> = ({ message, type, link }) => {
  if (type === 'title') {
    return <h1>{message}</h1>
  }

  if (type === 'subtitle') {
    return <h4>{message}</h4>
  }

  if (type === 'link' && link) {
    return <Link link={link} message={message}/>
  }

  return <p >{message}</p>
}

Text.defaultProps = {
  type: 'p'
} as Partial<TextProps>

export default Text;
