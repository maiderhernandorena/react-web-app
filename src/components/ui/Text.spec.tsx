import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Text from './Text';

describe('Text Component', () => {

  describe('Snapshots', () => {
    it('Should render normal text', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Text message='message' />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render title type text', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Text message='message' type='title'  />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render title type text', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Text message='message' type='subtitle'  />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('Should render link type text', () => {
      const shallowRenderer = ShallowRenderer.createRenderer();
      const wrapper = shallowRenderer.render(
        <Text message='message' type='link' link='google.com' />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});