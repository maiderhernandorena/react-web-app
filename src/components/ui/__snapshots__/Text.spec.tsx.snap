// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component Snapshots Should render link type text 1`] = `
<Link
  link="google.com"
  message="message"
/>
`;

exports[`Text Component Snapshots Should render normal text 1`] = `
<p>
  message
</p>
`;

exports[`Text Component Snapshots Should render title type text 1`] = `
<h1>
  message
</h1>
`;

exports[`Text Component Snapshots Should render title type text 2`] = `
<h4>
  message
</h4>
`;
